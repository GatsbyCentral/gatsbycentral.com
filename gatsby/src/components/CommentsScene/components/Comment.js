import React from "react";
import styled from "styled-components";
import moment from "moment";

const Comment = styled.div`
  border-bottom: 1px solid silver;
  padding: 1rem 0;
  font-size: 0.75rem;
  margin-bottom: 1rem;
`;
const Message = styled.div`
  font-size: 1rem;
  font-weight: 400;
`;
const Author = styled.span`
  font-weight: 800;
`;
const Date = styled.span``;

const Element = ({ comment: { node: comment } }) => {
  return (
    <Comment>
      <Message>{comment.message}</Message>
      {/* Date is `timestamp-seconds` format. Convert with `moment.unix()` */}
      <Date>{moment.unix(comment.date).format("LLL")}</Date> by{" "}
      <Author>{comment.name}</Author>
    </Comment>
  );
};

export default Element;
