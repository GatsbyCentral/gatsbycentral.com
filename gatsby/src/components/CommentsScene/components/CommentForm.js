import React from "react";
import styled from "styled-components";

const Form = styled.form`
  margin-top: 1rem;
  display: flex;
  flex-flow: wrap column;
`;

const Input = styled.input`
  font-size: 1rem;
  padding: 0.2rem;
  width: 100%;
  line-height: 2rem;
  margin-bottom: 1rem;
  border: 1px solid rgb(238, 238, 238);
`;

const Textarea = styled.textarea`
  font-size: 1rem;
  padding: 0.5rem;
  width: 100%;
  line-height: 2rem;
  border-color: rgb(238, 238, 238);
`;

const Button = styled.button`
  background-color: ${({ theme }) => theme.primaryColor};
  border: 2px solid ${({ theme }) => theme.primaryColor};
  font-weight: bold;
  font-size: 1rem;
  color: white;
  padding: 0.5rem;
  width: 30%;
  line-height: 2rem;
`;

// Use the `path` (frontmatter) as the identifier for the post.
const CommentForm = ({ postId }) => {
  return (
    <Form
      method="POST"
      action="https://api.staticman.net/v2/entry/GatsbyCentral/gatsbycentral.com/master/comments"
    >
      <input
        name="options[redirect]"
        type="hidden"
        value="https://www.gatsbycentral.com/comments/success"
      />
      <input name="fields[postPath]" type="hidden" value={postId} />
      <label>
        <Input name="fields[name]" type="text" placeholder="Name" required />
      </label>
      <label>
        <Input name="fields[email]" type="email" placeholder="eMail" required />
      </label>
      <label>
        <Textarea name="fields[message]" required />
      </label>

      <Button type="submit">Comment</Button>
    </Form>
  );
};

export default CommentForm;
