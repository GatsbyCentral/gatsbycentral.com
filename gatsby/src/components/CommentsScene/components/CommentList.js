import React from "react";

const CommentList = ({ comments }) => {
  return <div>comment</div>;
};

export default CommentList;
