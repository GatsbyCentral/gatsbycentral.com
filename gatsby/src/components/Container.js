import React from "react";
import styled from "styled-components";

const Container = styled.div`
  flex: 1;
  margin-left: 5px;
  margin-right: 5px;
`;

export default Container;
