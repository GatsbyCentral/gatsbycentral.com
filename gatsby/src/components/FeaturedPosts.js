import React from "react";
import styled from "styled-components";
import { map } from "ramda";
import Link from "gatsby-link";

const Title = styled.h2`
  text-align: center;
  border: none;
`;

const Posts = styled.div`
  text-align: center;
  font-size: 2rem;
  line-height: 4rem;
  a {
    //color: rgb(102, 102, 102);
  }
`;

const renderPosts = map(post => (
  <div key={post.node.id}>
    <Link to={post.node.frontmatter.path}>{post.node.frontmatter.title}</Link>
  </div>
));

const FeaturedPosts = props => {
  return (
    <div>
      <Title>Get Started</Title>
      <Posts>
        <div>
          <Link to="/what-is-gatsby-exactly">What is Gatsby exactly?</Link>
        </div>
        <div>
          <Link to="/introduction-to-gatsby-data-flow">
            Introduction to Gatsby data flow
          </Link>
        </div>
        <div>
          <Link to="/meta-tags-in-gatsby">Meta Tags in Gatsby</Link>
        </div>
      </Posts>
    </div>
  );
};

export default FeaturedPosts;
