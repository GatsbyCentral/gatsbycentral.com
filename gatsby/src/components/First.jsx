import React from 'react';
import Link from 'gatsby-link';
import Helmet from 'react-helmet';

import image_0 from '../images/pic01.jpg';

export default function First(props) {
 return (
   <section id="one" className="main style1">
     <div className="container">
       <div className="row 150%">
         <div className="6u 12u$(medium) custom-medium">
           <header className="major">
             <h2>Lorem ipsum dolor adipiscing<br />
             amet dolor consequat</h2>
           </header>
           <p>Adipiscing a commodo ante nunc accumsan et interdum mi ante adipiscing. A nunc lobortis non nisl amet vis sed volutpat aclacus nascetur ac non. Lorem curae et ante amet sapien sed tempus adipiscing id accumsan.</p>
         </div>
         <div className="6u 12u$(medium) important(medium) custom-medium">
           <span className="image fit"><img src={image_0} alt="" /></span>
         </div>
       </div>
     </div>
   </section>
 )
}
