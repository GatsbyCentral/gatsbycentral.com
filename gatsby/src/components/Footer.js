import React from "react";
import styled from "styled-components";
import Twitter from "images/twitter.svg";
import Facebook from "images/facebook.svg";
import Github from "images/github.svg";
import Meetup from "images/meetup.svg";

const Element = styled.div`
  margin-top: 2rem;
  color: white;
  background-color: ${({ theme }) => theme.secondaryColor};
  padding: 0.5rem;
  padding-top: 1rem;
`;

const Content = styled.div`
  max-width: 900px;
  margin: auto;
`;

const Link = styled.a`
  color: white;
  text-decoration: none;
`;

const Right = styled.div`
  float: right;
`;

const List = styled.ul`
  display: inline;
  list-style-type: none;
`;

const Item = styled.li`
  display: inline;
  margin-right: 1rem;
  color: white;
  width: 1.5rem;
  height: 1.5rem;
  a {
    color: white;
    svg {
      width: 1.3rem;
      height: 1.3rem;
    }
  }
`;

const Footer = props => {
  return (
    <Element>
      <Content>
        &copy;{" "}
        <Link href="https://superlumen.io" target="_blank">
          superlumen.io
        </Link>
        <Right>
          <List>
            <Item>
              <a href="https://twitter.com/gatsbycentral" target="_blank">
                <Twitter />
              </a>
            </Item>
            <Item>
              <a href="https://www.facebook.com/gatsbycentral/" target="_blank">
                <Facebook />
              </a>
            </Item>
            <Item>
              <a href="https://github.com/gatsbycentral" target="_blank">
                <Github />
              </a>
            </Item>
            <Item>
              <a href="https://www.meetup.com/gatsbyjs/" target="_blank">
                <Meetup />
              </a>
            </Item>
          </List>
        </Right>
      </Content>
    </Element>
  );
};

export default Footer;
