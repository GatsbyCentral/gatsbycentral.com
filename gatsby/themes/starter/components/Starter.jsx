import React from 'react';

export function Starter() {

  return (
    <div id="main">
      <p>
        Hello from Starter.
      </p>
    </div>
  );
}
