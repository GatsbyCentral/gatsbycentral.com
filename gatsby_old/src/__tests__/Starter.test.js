import React from 'react';
import renderer from 'react-test-renderer';
import { Starter } from '../components/Starter';

describe('<Starter />', () => {
  const component = mount(
    <Starter />
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

});
