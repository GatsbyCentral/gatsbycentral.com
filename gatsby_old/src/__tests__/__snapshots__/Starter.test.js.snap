// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Starter /> should render correctly 1`] = `
<Starter>
  <div
    id="main"
  >
    <p>
      Hello from Starter.
    </p>
  </div>
</Starter>
`;
